[xlsxio] new port

playgithub opened the pull request:
**Describe the pull request**

- #### What does your PR fix?  
Add port: xlsxio, read & write .xlsx files

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
All
No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

Commit:
add port: xlsxio

Commit:
vcpkg x-add-version xlsxio

playgithub commented:
I forked a barely new repo, and add port `xlsxio`, `vcpkg install -triplet:x64-windows` succeeds, I have no idea why it failed in CI.

Commit:
fix deps

JonLiu1993 commented on ports/xlsxio/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+)
+
+vcpkg_install_cmake()
+
```
Please add "vcpkg_fixup_pkgconfig()" here

playgithub replied:
ok

JonLiu1993 commented on ports/xlsxio/vcpkg.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "xlsxio",
+  "version-string": "commit-e3acace39e5fb153f5ce3500a4952c2bf93175bd",
```
Please use the version instead commit id

playgithub replied:
the latest commit is the best, but without a version number

JonLiu1993 replied:
You can replace the commit here with the date of the day of submission, and replace "version-string" with "version-date"

playgithub replied:
There might be many commits in one day, I think commit id is better.
While using date, it just looks good.

JackBoosY replied:
For commit id, please use `version-date`, see [docs](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#examples-1).

playgithub replied:
If there are multiple commits in one day, date as version is not enough.

JackBoosY replied:
@playgithub This is not important because it is impossible to update the port multiple times in the same day in vcpkg.

JonLiu1993 commented on ports/xlsxio/vcpkg.json:
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "xlsxio",
+  "version-string": "commit-e3acace39e5fb153f5ce3500a4952c2bf93175bd",
+  "port-version": 1,
```
We should delete the "port-version"

playgithub replied:
Have `port-version` been deprycated?

JonLiu1993 replied:
When adding a new port, port-version needs to be set to 0, so we usually do not add port-version

playgithub replied:
OK
A UI assistant will make a vcpkg contributor much easier, maybe I'll do it later.

JonLiu1993 commented:
Please take a look:
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6654356/install-x86-windows-dbg-out.log)

playgithub commented:
> Please take a look:
> [install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/6654356/install-x86-windows-dbg-out.log)

It should a bug of port `minizip`, https://github.com/microsoft/vcpkg/issues/18416

Commit:
add vcpkg_fixup_pkgconfig()

playgithub commented on ports/xlsxio/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
+                  REPO brechtsanders/xlsxio
+                  REF e3acace39e5fb153f5ce3500a4952c2bf93175bd
+                  SHA512 8148b89c43cf45653c583d51fb8050714d3cd0a76ab9a05d46604f3671a06487e4fc58d3f6f9f2a9f9b57a9f9fe1863ef07017c74197f151390576c5aac360ea
+                  HEAD_REF master)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+                     FEATURES
+                        libzip WITH_LIBZIP
+                        wide WITH_WIDE)
```
`vcpkg_from_github` or `vcpkg_check_features`?
I have double checked them, they seem ok.

JonLiu1993 replied:
What I mean is that the parameters in the two functions need four spaces from the beginning of the line

playgithub replied:
`vcpkg format-manifest` is for `vcpkg.json`
Is there a rule for `portfile.cmake`?

JonLiu1993 replied:
We have no clear rules to restrict the format of portiflile.cmake, but it is best to write the specifications as in the example in the official documentation

playgithub replied:
I think the format in the doc is more consistent  when it's more convenient to comment out some lines.

JackBoosY commented on ports/xlsxio/portfile.cmake:
```diff
@@ -0,0 +1,39 @@
+vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
+                  REPO brechtsanders/xlsxio
+                  REF e3acace39e5fb153f5ce3500a4952c2bf93175bd
+                  SHA512 8148b89c43cf45653c583d51fb8050714d3cd0a76ab9a05d46604f3671a06487e4fc58d3f6f9f2a9f9b57a9f9fe1863ef07017c74197f151390576c5aac360ea
+                  HEAD_REF master)
```
```suggestion
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO brechtsanders/xlsxio
    REF e3acace39e5fb153f5ce3500a4952c2bf93175bd
    SHA512 8148b89c43cf45653c583d51fb8050714d3cd0a76ab9a05d46604f3671a06487e4fc58d3f6f9f2a9f9b57a9f9fe1863ef07017c74197f151390576c5aac360ea
    HEAD_REF master
)
```

JackBoosY commented on ports/xlsxio/portfile.cmake:
```diff
@@ -0,0 +1,39 @@
+                  SHA512 8148b89c43cf45653c583d51fb8050714d3cd0a76ab9a05d46604f3671a06487e4fc58d3f6f9f2a9f9b57a9f9fe1863ef07017c74197f151390576c5aac360ea
+                  HEAD_REF master)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+                     FEATURES
+                        libzip WITH_LIBZIP
+                        wide WITH_WIDE)
```
```suggestion
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        libzip WITH_LIBZIP
        wide WITH_WIDE
)
```

playgithub committed:
Apply suggestions from code review

To be consistent, `OUT_FEATURE_OPTIONS FEATURE_OPTIONS` in a new line, as in `vcpkg_from_git`.

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

playgithub committed:
Update ports/xlsxio/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

Commit:
format

Commit:
use date as version string

Commit:
vcpkg format-manifest

playgithub commented:
@JackBoosY 
With https://github.com/microsoft/vcpkg/pull/18472, it still got the same problem.

JackBoosY commented:
@playgithub  Don't worry I will handle that.

JackBoosY commented:
/azp run

playgithub reviewed (commented):
> /azp run

What does it mean?

playgithub commented:
> > /azp run
> 
> What does it mean?

OK, it is a command.

JackBoosY commented:
Sadly, I don't have permission to push my changes to this branch.

JackBoosY commented on ports/xlsxio/portfile.cmake:
```diff
@@ -0,0 +1,43 @@
+    REF e3acace39e5fb153f5ce3500a4952c2bf93175bd
+    SHA512 8148b89c43cf45653c583d51fb8050714d3cd0a76ab9a05d46604f3671a06487e4fc58d3f6f9f2a9f9b57a9f9fe1863ef07017c74197f151390576c5aac360ea
+    HEAD_REF master
+)
```
```suggestion
    PATCHES fix-dependencies.patch
)

file(REMOVE ${SOURCE_PATH}/CMake/FindMinizip.cmake)
```

Please unzip [this patch](https://github.com/microsoft/vcpkg/files/6674188/fix-dependencies.zip), put it into _ports/xlsxio_ then commit it.

playgithub committed:
Update ports/xlsxio/portfile.cmake

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

Commit:
patch for dependency

JackBoosY commented:
Sorry, please replace the patch with [this](https://github.com/microsoft/vcpkg/files/6674278/fix-dependencies.zip).

playgithub commented:
@JackBoosY 
It passes the test for x64_windows, but failed in other cases.

JackBoosY commented:
@playgithub  Yeah, please apply my suggestion in https://github.com/microsoft/vcpkg/pull/18438#issuecomment-863713901.

Commit:
fix typo

playgithub commented:
> @playgithub Yeah, please apply my suggestion in [#18438 (comment)](https://github.com/microsoft/vcpkg/pull/18438#issuecomment-863713901).

Now one error left.

JackBoosY reviewed (changes requested):
Please run `vcpkg x-add-version --overwrite-version xlsxio` then commit changes.

Commit:
vcpkg x-add-version

JackBoosY commented on versions/x-/xlsxio.json:
```diff
@@ -0,0 +1,14 @@
+      "git-tree": "5c7dbcf7085056b04fd949999e50162303a043b6",
+      "version-string": "2021-03-24+0100",
+      "port-version": 0
+    },
+    {
+      "git-tree": "70b37575e5bc5191e2522cde8b783d20f2afc1e7",
+      "version-string": "commit-e3acace39e5fb153f5ce3500a4952c2bf93175bd",
+      "port-version": 1
+    }
```
```suggestion
    }
```

ras0219-msft replied:
PRs must add only one version to any library.

playgithub commented:
> Please run `vcpkg x-add-version --overwrite-version xlsxio` then commit changes.

Done, it is not an easy work, thanks.
BTW, I'm learn cmake systematically, hope that will help later.

JonLiu1993 commented:
@playgithub ,I tested the features locally, but none of the following triplet tests passed
- x86-windows
- x64-windows
- x64-windows-static
This is error log,please take a look:

config-x86-windows-out.log
````
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
EXPATW_LIBRARIES
    linked by target "xlsxio_readw_SHARED" in directory F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d

-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_LIBDIR
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
[2/2] cmd /c "cd .. && "F:/test-pr/3/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe" "F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d" "-DWITH_LIBZIP=ON" "-DWITH_WIDE=ON" "-DBUILD_SHARED=ON" "-DBUILD_STATIC=OFF" "-DBUILD_DOCUMENTATION=OFF" "-DBUILD_EXAMPLES=OFF" "-DBUILD_PC_FILES=OFF" "-DBUILD_TOOLS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/test-pr/3/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/test-pr/3/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=F:/test-pr/3/vcpkg/packages/xlsxio_x86-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "F:/test-pr/3/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe" "F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d" "-DWITH_LIBZIP=ON" "-DWITH_WIDE=ON" "-DBUILD_SHARED=ON" "-DBUILD_STATIC=OFF" "-DBUILD_DOCUMENTATION=OFF" "-DBUILD_EXAMPLES=OFF" "-DBUILD_PC_FILES=OFF" "-DBUILD_TOOLS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/test-pr/3/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/test-pr/3/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=F:/test-pr/3/vcpkg/packages/xlsxio_x86-windows""
CMake Deprecation Warning at CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- The C compiler identification is MSVC 19.28.29914.0
-- The CXX compiler identification is MSVC 19.28.29914.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/zlib.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/zlibd.lib (found version "1.2.11") 
-- Found LibZip: F:/test-pr/3/vcpkg/installed/x86-windows/lib/zip.lib;optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/zlib.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/zlibd.lib  
-- Found EXPAT: optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/libexpat.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/libexpatd.lib (found version "2.4.1") 
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Configuring done
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
EXPATW_LIBRARIES
    linked by target "xlsxio_readw_SHARED" in directory F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d

-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_LIBDIR
    _VCPKG_ROOT_DIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
ninja: build stopped: subcommand failed.
````

playgithub commented:
> @playgithub ,I tested the features locally, but none of the following triplet tests passed
> 
> * x86-windows
> * x64-windows
> * x64-windows-static
>   This is error log,please take a look:
> 
> config-x86-windows-out.log
> 
> ```
> CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
> Please set them or make sure they are set and tested correctly in the CMake files:
> EXPATW_LIBRARIES
>     linked by target "xlsxio_readw_SHARED" in directory F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d
> 
> -- Generating done
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     CMAKE_INSTALL_BINDIR
>     CMAKE_INSTALL_LIBDIR
>     _VCPKG_ROOT_DIR
> 
> 
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> [2/2] cmd /c "cd .. && "F:/test-pr/3/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe" "F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d" "-DWITH_LIBZIP=ON" "-DWITH_WIDE=ON" "-DBUILD_SHARED=ON" "-DBUILD_STATIC=OFF" "-DBUILD_DOCUMENTATION=OFF" "-DBUILD_EXAMPLES=OFF" "-DBUILD_PC_FILES=OFF" "-DBUILD_TOOLS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/test-pr/3/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/test-pr/3/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=F:/test-pr/3/vcpkg/packages/xlsxio_x86-windows""
> FAILED: ../CMakeCache.txt 
> cmd /c "cd .. && "F:/test-pr/3/vcpkg/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/bin/cmake.exe" "F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d" "-DWITH_LIBZIP=ON" "-DWITH_WIDE=ON" "-DBUILD_SHARED=ON" "-DBUILD_STATIC=OFF" "-DBUILD_DOCUMENTATION=OFF" "-DBUILD_EXAMPLES=OFF" "-DBUILD_PC_FILES=OFF" "-DBUILD_TOOLS=OFF" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v142" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=F:/test-pr/3/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=F:/test-pr/3/vcpkg" "-D_VCPKG_INSTALLED_DIR=F:/test-pr/3/vcpkg/installed" "-DVCPKG_MANIFEST_INSTALL=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=F:/test-pr/3/vcpkg/packages/xlsxio_x86-windows""
> CMake Deprecation Warning at CMakeLists.txt:1 (CMAKE_MINIMUM_REQUIRED):
>   Compatibility with CMake < 2.8.12 will be removed from a future version of
>   CMake.
> 
>   Update the VERSION argument <min> value or use a ...<max> suffix to tell
>   CMake that the project does not need compatibility with older versions.
> 
> 
> -- The C compiler identification is MSVC 19.28.29914.0
> -- The CXX compiler identification is MSVC 19.28.29914.0
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/cl.exe - skipped
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Enterprise/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x86/cl.exe - skipped
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Found ZLIB: optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/zlib.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/zlibd.lib (found version "1.2.11") 
> -- Found LibZip: F:/test-pr/3/vcpkg/installed/x86-windows/lib/zip.lib;optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/zlib.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/zlibd.lib  
> -- Found EXPAT: optimized;F:/test-pr/3/vcpkg/installed/x86-windows/lib/libexpat.lib;debug;F:/test-pr/3/vcpkg/installed/x86-windows/debug/lib/libexpatd.lib (found version "2.4.1") 
> -- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
> -- Configuring done
> CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
> Please set them or make sure they are set and tested correctly in the CMake files:
> EXPATW_LIBRARIES
>     linked by target "xlsxio_readw_SHARED" in directory F:/test-pr/3/vcpkg/buildtrees/xlsxio/src/2bf93175bd-948250290d
> 
> -- Generating done
> CMake Warning:
>   Manually-specified variables were not used by the project:
> 
>     CMAKE_INSTALL_BINDIR
>     CMAKE_INSTALL_LIBDIR
>     _VCPKG_ROOT_DIR
> 
> 
> CMake Generate step failed.  Build files cannot be regenerated correctly.
> ninja: build stopped: subcommand failed.
> ```

Does it failed when feature `wide` is on?

JackBoosY commented:
If the featrure `wide` needs `expat[wide]`, I suggest you to disable this feature now, and add it in the future when expat also support wide char.

Commit:
remove feature wide

ras0219-msft commented on ports/xlsxio/vcpkg.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "xlsxio",
+  "version-string": "2021-03-24+0100",
```
Consider using `version-date` instead:
```suggestion
  "version-date": "2021-03-24",
```

playgithub replied:
Is there a default time zone?
Is `version-date` something like a hint?

ras0219-msft replied:
There is not a default time zone, but this is not intended to be that granular. It is really only intended for ordering and rough suggestion of when the port was last changed.

ras0219-msft commented on ports/xlsxio/vcpkg.json:
```diff
@@ -0,0 +1,28 @@
+        "bzip2"
+      ]
+    }
+  ],
+  "features": {
+    "libzip": {
+      "description": "use libzip instead of minizip",
+      "dependencies": [
+        {
+          "name": "libzip",
+          "features": [
+            "bzip2"
+          ]
+        }
+      ]
+    }
+  }
```
```suggestion
  ]
```

Features cannot be used to model replacements like "Use X instead of Y" -- features must only be additive.

In this case, we must choose the most reasonable default (minizip seems fine) and users that want the other configuration can use overlay ports.

playgithub replied:
It's ok, the author also suggests `minizip`, and `libzip` causes a bug that libreoffice can't open xlsx files that edited when using `libzip`.

ras0219-msft reviewed (changes requested):
Thanks for the PR!

playgithub committed:
Update ports/xlsxio/vcpkg.json

deprecate libzip that's an alternative and imperfect

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

playgithub committed:
Update ports/xlsxio/vcpkg.json

Co-authored-by: Robert Schumacher <roschuma@microsoft.com>

JackBoosY commented on versions/x-/xlsxio.json:
```diff
@@ -0,0 +1,19 @@
+      "git-tree": "363ef11ba53c472f8ec8186c455f964b49088d7f",
+      "version-date": "2021-03-24",
+      "port-version": 0
+    },
+    {
+      "git-tree": "f38380a74a82c51f78ecb3f60ec56e0432631b90",
+      "version-string": "2021-03-24+0100",
+      "port-version": 0
+    },
+    {
+      "git-tree": "70b37575e5bc5191e2522cde8b783d20f2afc1e7",
+      "version-string": "commit-e3acace39e5fb153f5ce3500a4952c2bf93175bd",
+      "port-version": 1
+    }
```
```suggestion
    }
```

playgithub replied:
BTW, I have used `vcpkg x-add-verison` as the last commit, I don't known why it is right.

playgithub committed:
Update versions/x-/xlsxio.json

Co-authored-by: Jack·Boos·Yu <47264268+JackBoosY@users.noreply.github.com>

ras0219-msft committed:
Remove x64-windows-cpp17 community triplet

ras0219-msft reviewed (approved):
Thanks for the PR!

playgithub reviewed (commented):
The triplet `x64-windows-cpp17` is there by accident, should be deleted.

playgithub commented:
The errors on osx seems irrelevant. The port had passed all the CI tests before removing a custom triplet, it looks like some changes of the CI system caused the error. I have no idea.
```
2021-06-25T05:11:11.3015480Z     Working Directory: /Users/vagrant/Data/buildtrees/simage/x64-osx-dbg
2021-06-25T05:11:11.3016100Z     Error code: 1
2021-06-25T05:11:11.3016610Z     See logs for more information:
2021-06-25T05:11:11.3017480Z       /Users/vagrant/Data/buildtrees/simage/config-x64-osx-dbg-out.log
2021-06-25T05:11:11.3018390Z       /Users/vagrant/Data/buildtrees/simage/config-x64-osx-dbg-err.log
2021-06-25T05:11:11.3018810Z 
2021-06-25T05:11:11.3019170Z Call Stack (most recent call first):
2021-06-25T05:11:11.3019760Z   scripts/cmake/vcpkg_configure_cmake.cmake:332 (vcpkg_execute_required_process)
2021-06-25T05:11:11.3020420Z   ports/simage/portfile.cmake:36 (vcpkg_configure_cmake)
2021-06-25T05:11:11.3020860Z   scripts/ports.cmake:139 (include)
2021-06-25T05:11:11.3021170Z 
2021-06-25T05:11:11.3021340Z 
2021-06-25T05:11:11.3021960Z Error: Building package simage:x64-osx failed with: BUILD_FAILED
2021-06-25T05:11:11.3022860Z Elapsed time for package simage:x64-osx: 5.119 s
2021-06-25T05:11:11.3023500Z Starting package 143/173: coin:x64-osx
2021-06-25T05:11:11.3024320Z Building package coin[core,simage,zlib]:x64-osx...
2021-06-25T05:11:11.3025260Z Error: Building package coin:x64-osx failed with: CASCADED_DUE_TO_MISSING_DEPENDENCIES
2021-06-25T05:11:11.3026180Z Elapsed time for package coin:x64-osx: 19 us
```

dg0yt commented:
The osx trouble with simage is a different story: A possible issue with tiff's cmake config wrapper in combination with simage dependency configuration in connection with a vcpkg `find_package` issue (https://github.com/microsoft/vcpkg/issues/18631).  There are PRs pending for tiff and simage.

playgithub commented:
> The osx trouble with simage is a different story: A possible issue with tiff's cmake config wrapper in combination with simage dependency configuration in connection with a vcpkg `find_package` issue (#18631). There are PRs pending for tiff and simage.

Thanks
Also `coin` and some that are not pasted.
BTW I have switched to Linux from Windows, libs and build tools are supported natively on Linux, it never blocks my work.

JackBoosY commented:
Waiting for other PR to fix the simage regression.

playgithub commented:
/azp run

playgithub commented:
@JackBoosY 
#18645 is merge, please `/azp run`

JonLiu1993 commented:
/azp run

playgithub commented:
@JonLiu1993 
In my experience, it can parse `.xlsx files` than `xlnt` can not. Why not merge it?

JonLiu1993 commented:
Ping @ras0219 @ras0219-msft, please help merge this PR, thanks!

ras0219-msft commented:
This LGTM, thanks for the new port!
